src/stdafx.h

```diff
@@ -418,21 +418,32 @@ assert_compile(SIZE_MAX >= UINT32_MAX);
 #	define CloseConnection OTTD_CloseConnection
 #endif /* __APPLE__ */
 
+#ifdef __GNUC__
+#	define likely(x)   __builtin_expect(!!(x), 1)
+#	define unlikely(x) __builtin_expect(!!(x), 0)
+#else
+#	define likely(x)   (x)
+#	define unlikely(x) (x)
+#endif
+
 void NORETURN CDECL usererror(const char *str, ...) WARN_FORMAT(1, 2);
 void NORETURN CDECL error(const char *str, ...) WARN_FORMAT(1, 2);
 void NORETURN CDECL assert_msg_error(int line, const char *file, const char *expr, const char *str, ...) WARN_FORMAT(4, 5);
 #define NOT_REACHED() error("NOT_REACHED triggered at line %i of %s", __LINE__, __FILE__)
 
-/* For non-debug builds with assertions enabled use the special assertion handler. */
-#if defined(NDEBUG) && defined(WITH_ASSERT)
+/* For non-debug builds with assertions enabled use the special assertion handler:
+ * - For MSVC: NDEBUG is set for all release builds and WITH_ASSERT overrides the disabling of asserts.
+ * - For non MSVC: NDEBUG is set when assertions are disables, _DEBUG is set for non-release builds.
```

Comment:
This is no longer what `cmake/Options.cmake` suggests. It used to be true once, I remember, so maybe this happens because you backported really old patches, but .. it feels weird now :)

Yeah, had to double-check, with the introduction of CMake this changed:

- For Debug builds with `WITH_ASSERT`, we just `assert()`
- For Release builds with `WITH_ASSERT`, we use `error()`
- For Release builds without `WITH_ASSERT`, we just ignore assert.

`NDEBUG` is set with Release build.

For non-beta, non-RC releases we normally set the `WITH_ASSERT` to `OFF`; for all other builds it remains on.